compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -2352,6 +2423,51 @@ impl Type {
             _ => None,
         }
     }
+
+    pub(crate) fn integral_maximum_size(&self) -> Option<FieldElement> {
+        match self {
+            Type::FieldElement => None,
+            Type::Integer(sign, num_bits) => {
+                let mut max_bit_size = num_bits.bit_size();
+                if sign == &Signedness::Signed {
+                    max_bit_size >>= 1;
+                }
+                Some(((1u128 << max_bit_size) - 1).into())
```

Comment:
Shouldn't the signed case be `max_bit_size -= 1`?
We want to divide the total end value by 2 but this is just the bit count